core/src/main/java/org/apache/calcite/sql/validate/SqlValidatorImpl.java

```diff
@@ -3039,6 +3042,14 @@ protected void validateOver(SqlCall call, SqlValidatorScope scope) {
     throw new AssertionError("OVER unexpected in this context");
   }
 
+  protected void validateUnnest(SqlCall call, SqlValidatorScope scope, RelDataType targetRowType) {
+    for (int i = 0; i < call.operandCount(); i++) {
+      SqlNode expandedItem = expand(call.operand(i), scope);
+      call.setOperand(i++, expandedItem);
```

Comment:
Not clear why it is required one more increment here. May be add an extra comment

Reply:
Sorry, I forget to remove it when replaced for each loop with for i. There shouldn't be one more increment. Thanks for pointing this.